### ISSUE TITLE
question: trigger a new CI build on commit

### ISSUE THREAD

**Issue by rokroskar** [stefanzweifel/git-auto-commit-action#38]
Hi there, thanks for this great little github action! We use it to do some automatic cleanup of commits from another bot. I'm in a situation now where I have to push another commit manually in order to be able to merge the PR because this action doesn't trigger another CI build so the last commit on the branch doesn't have the required status checks. Is there some way to circumvent this? I guess I could have a job follow this action to trigger a new build?

**Comment by localheinz** [comment]
@rokroskar 

You could 

- create an account that is used for these kinds of actions
- create a personal access token for that account
- use that personal access token instead of the `GITHUB_TOKEN`

This is what I'm doing with @ergebnis-bot, and it works very well.

**Comment by rokroskar** [comment]
Aha I think I'm missing something here - is it that pushes with the `GITHUB_TOKEN` don't trigger CI builds by default?

**Comment by localheinz** [comment]
@rokroskar 

That's it, yes!

**Comment by rokroskar** [comment]
~~Weird. I don't see this mentioned anywhere in the [docs](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token#permissions-for-the-github_token).~~

Ah, it is in fact [documented](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#about-workflow-events) - but in that case, I guess that using a different token won't matter, it would need to be triggered via an API call. Thanks @localheinz!

**Comment by stefanzweifel** [comment]
Thanks @localheinz for helping out here! 👍 

@rokroskar, can we close this issue? Could you solve your problem with Personal Access Tokens?

**Comment by rokroskar** [comment]
ah actually I wanted to follow up here - yes, I can use a personal access token. However, I think github checks the committer to determine whether to trigger the CI build or not. If you use the default `GITHUB_TOKEN`, the committer is automatically set to "github actions". If you use your personal access token, the committer is set to whatever user you use for that. *However* in this action, the committer is set explicitly to github actions here: https://github.com/stefanzweifel/git-auto-commit-action/blob/master/entrypoint.sh#L47-L48

This means that no matter which personal access token I use, I always commit as `actions@github.com` which means it will never trigger a CI build. A better solution to what is being done now would be to replace the login with something like this:

```
git remote set-url origin https://${GITHUB_TOKEN}@github.com/${GITHUB_REPOSITORY}.git 
```

This will ensure that the commit is always in the name of the `GITHUB_TOKEN` owner. I can try to make a PR if you want, but it's a simple change so maybe you prefer to handle it yourself?

**Comment by stefanzweifel** [comment]
Ah, I see.

Other users have mentioned this as well. We will add a config option to update the committer name/email very soon. (https://github.com/stefanzweifel/git-auto-commit-action/issues/29#issuecomment-574062126)

All these changes will haben in v3.0.0. I hope I can release  this in the next days. I will keep this issue open as a reminder for myself.
I will keep you posted, as soon as the issue has been resolved.

**Comment by rokroskar** [comment]
Yes, that would be a good addition as well, was going to mention it :) 

I just tried only setting the remote and indeed of course it isn't enough, you have to also provide the committer name and email. I was hoping this was inferred from the token on push but it isn't the case. Thanks!

**Comment by stefanzweifel** [comment]
@rokroskar Hi again, I've just released v3.0.0 of this Action. We've added new options which allow you update the commit user and author.

You can find usage examples [here](https://github.com/stefanzweifel/git-auto-commit-action/tree/v3.0.0#usage).
In addition, `actions/checkout@v2` is now required. So you no longer have to pass a `GITHUB_TOKEN` to this Action, but rather to `checkout`.

I didn't test, if these changes fix the issue described here. Maybe you have the time to do that? Would be great if we could solve this problem and add an example to the README for others.

**Comment by stefanzweifel** [comment]
Hi again,

I've just updated the README with a troubleshoot section, which describes how to solve this problem: https://github.com/stefanzweifel/git-auto-commit-action#commit-of-this-action-does-not-trigger-a-new-workflow-run

TLDR: Use a Personal Access Token instead of the default `GITHUB_TOKEN`.

---

I'm closing this issue now, as this problem is now properly documented. Feel free to open a new issue or reopen this one if you have further questions.
